### DISCUSSION TITLE
Clients can't access LAN

### DISCUSSION THREAD

**Discussion by Nelse74** [strongswan/strongswan#1124] [Q&A]
Hello everyone,

My intention is for authenticated clients to access the LAN resources after they successfully connect with the right subnet local_ts.
I've added the iptables rules from [https://docs.strongswan.org/docs/5.9/howtos/forwarding.html](url), but I can't find them using iptables -S.
The iptables rules I added : 
iptables -t nat -A POSTROUTING -s 172.16.10.0/24 -o ens4 -m policy --dir out --pol ipsec -j ACCEPT
iptables -t nat -A POSTROUTING -s 172.16.10.0/24 -o ens4 -j MASQUERADE
iptables -t nat -I POSTROUTING -m policy --pol ipsec --dir out -j ACCEPT

I also didn't install iptables-services, I'm thinking that just reloading the firewalld service makes it work (probably not since I can't see them in iptables -S)
Should I just install the iptables-services and run both of them at the same time?
My tutor wants me to use firewalld instead of iptables so I've added direct rules in firewalld that should do the work : 

```
<?xml version="1.0" encoding="utf-8"?>
<direct>
  <rule ipv="ipv4" table="nat" chain="POSTROUTING" priority="0">-o ens4 -j MASQUERADE</rule>
  <rule ipv="ipv4" table="filter" chain="FORWARD" priority="0">-i ens4 -o ens3 -j ACCEPT</rule>
  <rule ipv="ipv4" table="filter" chain="FORWARD" priority="0">-i ens4 -o ens3 -m state --state RELATED,ESTABLISHED -j ACCEPT</rule>
</direct>
```

firewall-cmd --list-all-zones : 

public (active)
  target: default
  icmp-block-inversion: no
  interfaces: ens4
  sources: 
  services: cockpit dhcpv6-client ipsec ssh
  ports: 500/udp 4500/udp
  protocols: 
  forward: no
  masquerade: yes
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules: 
	rule protocol value="esp" accept
	rule protocol value="ah" accept

internal (active)
  target: default
  icmp-block-inversion: no
  interfaces: ens3
  sources: 
  services: cockpit dhcpv6-client mdns samba-client ssh
  ports: 445/tcp
  protocols: 
  forward: no
  masquerade: yes
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules: 

ip of ens3 : 172.16.10.111/24
ip of ens4 : 178.X.X.X
Both of them have masquerading

When doing a tracert 172.16.10.111 on the client, it doesn't know where to go after its public gateway.

ip xfrm policy : 

src 172.16.10.0/24 dst 10.10.10.5/32 
	dir out priority 371327 ptype main 
	tmpl src 178.X.X.X dst 81.185.166.211
		proto esp spi 0xac412332 reqid 1 mode tunnel
src 10.10.10.5/32 dst 172.16.10.0/24 
	dir fwd priority 371327 ptype main 
	tmpl src 81.185.166.211 dst 178.X.X.X
		proto esp reqid 1 mode tunnel
src 10.10.10.5/32 dst 172.16.10.0/24 
	dir in priority 371327 ptype main 
	tmpl src 81.185.166.211 dst 178.X.X.X
		proto esp reqid 1 mode tunnel

ip route show table 220 : 

10.10.10.0 dev ens4 scope link (i've added it manually but since it's a virtual ip address in /32 I don't think it's right)
10.10.10.5 via 178.X.X.X (gateway of ens4) dev ens4 proto static src 172.16.10.111

sysctl -p : 

sysctl -p
net.ipv4.ip_forward = 1
net.ipv4.ip_no_pmtu_disc = 1
net.ipv4.conf.all.accept_redirects = 0
net.ipv4.conf.all.send_redirects = 0

With tcpdump -i ens4 I can see the connexion being made between the client and the vpn, but nothing after that.
What's funny is that the xfrm policies get dropped after some time, same goes for the route in table 220. Is there any reason for that like the rekeying of the CHILD_SA ?

I'm a bit new with routing and I don't know what to try next. It would be better if I only use firewalld and not iptables, my tutor isn't found of it.
My best guess is that when the traffic arrives to the public interface, it doesn't know where to go because i'm missing firewall rule... I feel like i'm messing with firewalld a lot.

Any tips or leads would be really appreciated ! Have a wonderful day

**Comment by tobiasbrunner** [comment]
> but I can't find them using iptables -S

That command lists rules in the `filter` table, by default. To see the rules in `nat` you have to pass `-t nat` (or just use `iptables-save`).

> I also didn't install iptables-services

No idea what that is. But I don't think it's necessary to just use `iptables`.

> so I've added direct rules in firewalld that should do the work :

I don't really know firewalld, but these are clearly not the same rules.

See #196.
